library/paths/bool.red

```diff
@@ -0,0 +1,38 @@
+import prelude
+import data.void
+import data.unit
+import data.bool
+import basics.isotoequiv
+
+def bool-path/code : bool → bool → type =
+  elim [
+  | tt → elim [
+    | tt → unit
+    | ff → void
+    ]
+  | ff → elim [
+    | tt → void
+    | ff → unit
+    ]
+  ]
+
+def bool-refl : (x : bool) → bool-path/code x x =
+  elim [
```

Comment:
I think this can be made even shorter by writing `\ * -> triv`; the idea is that `*` is a special "inversion pattern" which calls the eliminator, and then provides the rhs to each branch.